dds/DCPS/DataReaderImpl.h

```diff
@@ -88,6 +88,7 @@ class OpenDDS_Dcps_Export WriterStats {
   WriterStats(
     int amount = 0,
     DataCollector<double>::OnFull type = DataCollector<double>::KeepOldest);
+  WriterStats(const WriterStats&);
```

Comment:
Maybe also the assignment operator can be default and both implementations can be removed. With non C++11 compilers the compiler will also generate a default one.
``` suggestion
#if defined (ACE_HAS_CPP11)
  WriterStats(const WriterStats&) = default;
#endif
``` suggestion